tests/keras/test_keras_model_export.py

```diff
@@ -210,25 +210,34 @@ def _import_module(name, **kwargs):
             assert x == b
 
 
-@pytest.mark.skip("Unskip this test once https://github.com/h5py/h5py/issues/1732 is fixed")
-@pytest.mark.parametrize("build_model", [model, tf_keras_model])
+@pytest.mark.parametrize(
+    "build_model,save_format",
+    # TODO: Unskip these parameters once https://github.com/h5py/h5py/issues/1732 is fixed
+    # [(model, None), (tf_keras_model, None), (tf_keras_model, "h5"),
+    [(tf_keras_model, "tf")],
+)
 @pytest.mark.large
-def test_model_save_load(build_model, model_path, data):
+def test_model_save_load(build_model, save_format, model_path, data):
     x, _ = data
     keras_model = build_model(data)
     if build_model == tf_keras_model:
         model_path = os.path.join(model_path, "tf")
     else:
         model_path = os.path.join(model_path, "plain")
     expected = keras_model.predict(x)
-    mlflow.keras.save_model(keras_model, model_path)
+    kwargs = {"save_format": save_format} if save_format else {}
+    mlflow.keras.save_model(keras_model, model_path, **kwargs)
     # Loading Keras model
     model_loaded = mlflow.keras.load_model(model_path)
-    assert type(keras_model) == type(model_loaded)
-    assert all(expected == model_loaded.predict(x))
+    # When saving as SavedModel, we actually convert the model
+    # to a slightly different format, so we cannot assume it is
+    # exactly the same.
+    if save_format != "tf":
+        assert type(keras_model) == type(model_loaded)
+    assert expected == pytest.approx(model_loaded.predict(x))
     # Loading pyfunc model
     pyfunc_loaded = mlflow.pyfunc.load_model(model_path)
-    assert all(pyfunc_loaded.predict(x).values == expected)
+    assert expected == pytest.approx(pyfunc_loaded.predict(x).values)
```

Comment:
Can we use `numpy.testing.assert_array_almost_equal` instead?

Reply:
Good suggestion! Looking at the docstring on that, it suggests using something else for consistent handling; so I'm trying `np.testing.assert_allclose`. I used that setting relative tolerance to `1e-5`. Any stricter comparison and it will fail; if I'm honest, I expected slightly more tolerance (e.g. ~1e-6). But I'm not sure I have a good argument either way.

Do you think relative tolerance of `1e-5` is reasonable here?

Reply:
Looks reasonable :)